src/lib/y2storage/planned/can_be_encrypted.rb

```diff
@@ -102,7 +114,12 @@ def final_device!(plain_device)
         result = super
         if create_encryption?
           method = encryption_method || EncryptionMethod.find(:luks1)
-          result = plain_device.encrypt(method: method, password: encryption_password)
+          args = {}
+          if method.is?(:pervasive_luks2)
```

Comment:
Making this difference is indeed the only way to make it work. But that's IMHO due to a sub-optimal implementation of the classes handling pervasive encryption (my fault, I was too focused on the Partitioner case and didn't consider the planned devices).

So please add a comment so I will try to fix it in the future. Something like "FIXME: For pervasive_luks2 the arguments need to be passed directly at #encrypt instead of being able to assign them afterwards. That's a defect on the API of that encryption method that should be fixed."